server/src/main/java/io/crate/fdw/JdbcBatchIterator.java

```diff
@@ -168,7 +171,8 @@ public boolean moveNext() {
             if (resultSet.next()) {
                 for (int i = 0; i < columns.size(); i ++) {
                     Reference ref = columns.get(i);
-                    Object object = resultSet.getObject(i + 1);
+                    ResultSetMetaData resultSetMetaData = resultSet.getMetaData();
+                    Object object = getObject(resultSet, i, resultSetMetaData.getColumnTypeName(i + 1));
```

Comment:
I think the new `getObject` method is thoroughly tested by integration tests we already have.